### ISSUE TITLE
Improve querying with tags suggestion

### ISSUE THREAD

**Issue by jotak** [hawkular/hawkular-grafana-datasource#49]
Thanks to an enhancement on a the server side (https://issues.jboss.org/browse/HWKMETRICS-532), we can improve querying for metrics. I suggest:

- Removing the "query by id" vs "query by tags" selector and replace it with an unified querying panel
- This unified panel allows to provide both tags and metric name at the same time.
- Tags are now suggested thanks to HWKMETRICS-532
- Providing tags allows to refine metrics suggestion. Ie. if you have tons of metrics and it's difficult to scroll across all of them in suggestions, you can provide some tags to filter them
- You can provide tags only, omitting the metric name, in which case we would have the same behaviour as now with "query by tag".
- When a metric name is provided, tags are not used for querying. They're only present for easier metric suggestion
- In other words there's no change in the way metrics datapoints are fetched. Changes are only for metrics & tags suggestion in graph's editing panel.

**Comment by jotak** [comment]
This enhancement would be active for metrics >= 0.22.0. Current behaviour will still apply for earlier server versions.

**Comment by tsegismont** [comment]
Simplifying sounds good to me, but can you give an example of what a user
would type to differentiate a metric name from a metric tag if there's a
single text box?

2017-01-16 11:15 GMT+01:00 Joel Takvorian <notifications@github.com>:

**Comment by jotak** [comment]
My idea would be not to have a single text field, but maybe 2, on two lines: one for query, one for tag. I'm not totally decided yet on what to suggest... I'll post an example screenshot to illustrate.

**Comment by jotak** [comment]
(and of course it's open for discussion. I'll open a thread on the ml)

**Comment by jotak** [comment]
There's a new upcoming feature in Hawkular, a "tags query language" ( https://issues.jboss.org/browse/HWKMETRICS-523 ). I'll know more about it this afternoon, it may worth to integrate it somehow in Grafana.

**Comment by jotak** [comment]
replaced with #52
